includes/handle.inc

```diff
@@ -278,11 +280,39 @@ function islandora_handle_query_for_handle(AbstractObject $object) {
  * @return array
  *   An array describing the result of the database query.
  */
-function islandora_handle_retrieve_configurations_by_cmodels($models) {
+function islandora_handle_retrieve_configurations_by_cmodels(array $models) {
   $handle_results = db_select('islandora_handle_apply_handle', 'h')
     ->fields('h', array('cmodel', 'datastream', 'xsl_location'))
     ->condition('cmodel', $models)
     ->execute()
     ->fetchAll();
   return $handle_results;
 }
+
+/**
+ * Helper to determine whether the Handle has already been added in metadata.
+ *
+ * @param DOMDocument $alpha
+ *   The content of the datastream currently in DOMDocument form.
+ * @param DOMDocument $bravo
+ *   THe XML output of the transform.
+ *
+ * @return bool
+ *   TRUE if it exists, FALSE otherwise,
+ */
+function islandora_handle_datastream_content_equal(DOMDocument $alpha, DOMDocument $bravo) {
+  // Make a clone of the bravo DOMDocument to preserve whatever may have been
+  // transformed in the XSL from the Handle configuration.
+  $new_doc = new DOMDocument($bravo->version, $bravo->encoding);
+  $new_doc->appendChild($new_doc->importNode($bravo->documentElement->cloneNode(TRUE), TRUE));
```

Comment:
Something of a simplification... can drop the `cloneNode` call here (`importNode` copies on its own, into the importing document).